spec/60-trace-id-format.md

```diff
@@ -1,20 +1,52 @@
-`trace-id` ID on the left side.
-
-When a system operates with a `trace-id` that is shorter than 16 bytes, it
-SHOULD fill-in the extra bytes with random values rather than zeroes. Let's say
-the system works with an 8-byte `trace-id` like `3ce929d0e0e4736`. Instead of
-setting `trace-id` value to `0000000000000003ce929d0e0e4736` it SHOULD generate
-a value like `4bf92f3577b34da6a3ce929d0e0e4736` where `4bf92f3577b34da6a` is a
-random value or a function of time and host value.
-
-**Note**: Even though a system may operate with a shorter `trace-id` for
-[distributed trace](https://w3c.github.io/trace-context/#dfn-distributed-traces)
-reporting, the full `trace-id` MUST be propagated to conform to the
-specification.
+This section suggests some best practices to consider when platform or tracing
+vendor implement `trace-id` generation and propagation algorithms. These
+practices will ensure better interoperability of different systems.
+
+### Uniqueness of `trace-id`
+
+The value of `trace-id` SHOULD be globally unique. This field is typically used
+for unique identification of a <a>distributed trace</a>. It is common for
+<a>distributed traces</a> to span various components, including, for example,
+cloud services. Cloud services tend to serve variety of clients and have a very
+high throughput of requests. So global uniqueness of `trace-id` is important,
+even when local uniqueness might seem like a good solution.
+
+### Randomness of `trace-id`
+
+Randomly generated value of `trace-id` SDHOULD be preferred over other
+algorithms of generating a globally unique identifiers. Randomness of `trace-id`
+addresses some [secutiry](#security-considerations) and [privacy
+concerns](#privacy-considerations) of exposing unwanted information. Randomness
+also allows tracing vendors to base sampling decisions on `trace-id` field value
+and avoid propagating an additional sampling context.
+
+As shown in the next section, it is important for better inter-operability with
+some existing systems, for `trace-id` to carry "uniqueness" and "randomness" in
+the right part of the `trace-id`.
+
+### Length of `trace-id` for the platforms operating with shorter identifiers
+
+There are system operates with a `trace-id` that is shorter than 16 bytes,
+which still willing to adopt this specification.
+
+If such system capable of propagating an entire `trace-id`, even while operating
+internally on shorter identifier, system is encouraged to utilize the
+`tracestate` header to propagate shorter identifier. However, if system wants to
+use a subset of trace-id bytes as an identifier - rightmost part of a `trace-id`
+SHOULD be used.
+
+There are tracing systems that are not capable to propagate the entire 16 bytes
+of a `trace-id`. For better interoperability of these systems with the fully
+compliant systems, the following practices are recommended:
+
+1. Whenever a new `trace-id` needs to be generated - it SHOULD be left padded
```

Comment:
I think this whole point is irrelevant. How the system operates internally, including ID generation, is not important, because this specification would only matter on egress from such system.